test/ExecutionFramework.cpp

```diff
@@ -40,27 +42,51 @@ using namespace solidity::util;
 using namespace solidity::test;
 
 ExecutionFramework::ExecutionFramework():
-	ExecutionFramework(solidity::test::CommonOptions::get().evmVersion())
+	ExecutionFramework(solidity::test::CommonOptions::get().evmVersion(), solidity::test::CommonOptions::get().vmPaths)
 {
 }
 
-ExecutionFramework::ExecutionFramework(langutil::EVMVersion _evmVersion):
+ExecutionFramework::ExecutionFramework(langutil::EVMVersion _evmVersion, std::vector<boost::filesystem::path> const& _vmPaths):
 	m_evmVersion(_evmVersion),
 	m_optimiserSettings(solidity::frontend::OptimiserSettings::minimal()),
 	m_showMessages(solidity::test::CommonOptions::get().showMessages),
-	m_evmHost(make_shared<EVMHost>(m_evmVersion))
+	m_vmPaths(_vmPaths)
 {
 	if (solidity::test::CommonOptions::get().optimize)
 		m_optimiserSettings = solidity::frontend::OptimiserSettings::standard();
 
+	for (auto const& path: _vmPaths)
+	{
+		evmc::VM& vm = EVMHost::getVM(path.string());
+		if (!m_evmHost && vm.has_capability(EVMC_CAPABILITY_EVM1))
+			m_evmHost = make_shared<EVMHost>(m_evmVersion, vm);
+		if (!m_ewasmHost && vm.has_capability(EVMC_CAPABILITY_EWASM))
+			m_ewasmHost = make_shared<EVMHost>(m_evmVersion, vm);
+	}
+
+	switchVm();
+}
+
+void ExecutionFramework::switchVm(bool _useEwasm)
```

Comment:
Would it work to re-use the EVMC capabilities enum instead of the bool here?

Reply:
good idea, using `evmc_capabilities` now